src/tree/document/c14n.rs

```diff
@@ -0,0 +1,111 @@
+//! Document canonicalization logic
+//!
+use std::ffi::{c_int, c_void, CString};
+use std::os::raw;
+use std::ptr::null_mut;
+
+use crate::tree::c14n::*;
+
+use super::{
+  xmlAllocOutputBuffer, xmlC14NExecute, xmlC14NIsVisibleCallback, xmlChar, xmlNodePtr,
+  xmlOutputBufferClose, xmlOutputBufferPtr, Document,
+};
+
+impl Document {
+  /// Canonicalize a document and return the results.
```

Comment:
This docstring could use a more information or a link to the libxml2 documentation (which isn't that great either). The callback parameter is obvious and could benefit from an examples